master/lustrefactory.py

```diff
@@ -72,6 +83,54 @@ def buildCommand(props):
 
     return args
 
+def getTarballFactory(gerrit_repo, **kwargs):
+    """ Generates a build factory for a tarball generating builder.
+    Args:
+        gerrit_repo (string): Gerrit repo url
+    Returns:
+        BuildFactory: Build factory with steps for generating tarballs.
+    """
+    bf = util.BuildFactory()
+
+    # Pull the patch from Gerrit
+    bf.addStep(Gerrit(
+        repourl=gerrit_repo, workdir="build/lustre",
+        mode="full", method="fresh", retry=[60,60], timeout=3600,
+        logEnviron=False, getDescription=True, haltOnFailure=True,
+        description=["cloning"], descriptionDone=["cloned"]))
+
+    # make tarball
+    bf.addStep(ShellCommand(
+        command=['sh', './autogen.sh'],
+        description=["autogen"],
+        descriptionDone=["autogen"],
+        workdir="build/lustre"))
+
+    bf.addStep(Configure(
+        command=['./configure', '--enable-dist'],
+        workdir="build/lustre"))
+
+    bf.addStep(Compile(
+        command=['make', 'dist'],
+        workdir="build/lustre"))
+
+    # upload it to the master
+    bf.addStep(SetPropertyFromCommand(
+        command=['sh', '-c', 'ls *.tar.gz'],
+        property='tarball',
+        workdir="build/lustre",
+        hideStepIf=hide_except_error,
+        haltOnFailure=True))
+
+    bf.addStep(FileUpload(
+        workdir="build/lustre",
+        slavesrc=WithProperties("%(tarball)s"),
+        masterdest=WithProperties("public_html/tarballs/%(tarball)s"),
```

Comment:
Perhaps it doesn't need to be done in this patch, but eventually I think we want to organize the urls into a tree that consolidates the products from a gerrit change information.  For example:

 public_html/13353/2